python/cudf/cudf/_lib/column.pyx

```diff
@@ -538,7 +549,9 @@ cdef class Column:
                     owner=data_owner,
                     exposed=True,
                 )
-                if isinstance(data_owner, SpillableBuffer):
+                if isinstance(data_owner, CopyOnWriteBuffer):
+                    data_owner.ptr  # accessing the pointer marks it exposed.
```

Comment:
This looks wrong, `.ptr` is gone. Is this code path covered in tests, or am I completely missing something here?

Reply:
Yeah, I think this wants to be `data_owner.get_ptr(mode="write")` or (private API, bad) `data_owner._unlink_shared_buffers()`.

Reply:
Fixed it by adding a test for coverage.(test_dataframe_cow_slice_setitem)